TextileIPFSNativeModule.js

```diff
@@ -69,6 +79,26 @@ export default {
     return result
   },
 
+  registerToken: async function (hash: string, token: string) {
+    await TextileIPFS.setHashToken(hash, token)
+  },
+
+  getHashRequest: function (hash: string, path: string): HashRequest {
+    let token = ''
+    const chars = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789"
+    for (let i = 0; i < 48; i++) {
+      token += chars.charAt(Math.floor(Math.random() * chars.length))
+    }
+    this.registerToken(hash, token)
```

Comment:
Calling a async function here, no guarantee it will complete before the request returned from this function is made. I think as this is currently set up, we're basically hoping the async call will finish in time.

Reply:
yep, 100% the intention :)

Reply:
haha

Reply:
actually, just realized. in ios i can just make it sync, keep it sync in android for now

Reply:
done